src/workerd/api/global-scope.c++

```diff
@@ -477,28 +477,34 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketMessage(
 }
 
 void ServiceWorkerGlobalScope::sendHibernatableWebSocketClose(
-    kj::String reason,
     int code,
+    kj::String reason,
+    bool wasClean,
     Worker::Lock& lock, kj::Maybe<ExportedHandler&> exportedHandler) {
   auto event = jsg::alloc<HibernatableWebSocketEvent>();
 
   KJ_IF_MAYBE(h, exportedHandler) {
     KJ_IF_MAYBE(handler, h->webSocketClose) {
-      auto promise = (*handler)(lock, event->getWebSocket(lock), kj::mv(reason), code);
+      auto websocket = event->getWebSocket(lock);
+      websocket->initiateHibernatableRelease(lock, true);
```

Comment:
It's funny you mention that, I originally had this as an enum but switched it to a bool later because I figured someone would say it's better to use a bool than an enum here :sweat_smile:

Reply:
I moved the struct to another file because there was a circular dependency, but it should be fine https://github.com/cloudflare/workerd/pull/449/commits/a602115f236beb4fb35fc8293a72080f31abd082

Reply:
Okay, I think this is a bit cleaner now:
- https://github.com/cloudflare/workerd/pull/449/commits/3aa3ab1a1788a40e33d43d6830899ad68080f2af
- https://github.com/cloudflare/workerd/pull/449/commits/fbd4a5485f611505903f96b75fe087f4cea5e877